semantique/datacube.py

```diff
@@ -661,3 +663,320 @@ def _mask(self, data):
+                "specified spatio-temporal extent"
+            )
+        # Trim the array if requested.
+        # This will remove dimension coordinates with only missing or invalid data.
+        if self.config["trim"]:
+            data = data.sq.trim()
+        return data
+
+    def _load(self, metadata, extent):
+        # check if extent is valid
+        if TIME not in extent.dims:
+            raise exceptions.MissingDimensionError(
+                "Cannot retrieve data in an extent without a temporal dimension"
+            )
+        if X not in extent.dims and Y not in extent.dims:
+            raise exceptions.MissingDimensionError(
+                "Cannot retrieve data in an extent without spatial dimensions"
+            )
+
+        # retrieve spatial bounds, resolution & epsg
+        # note: round to avoid binary format <-> floating-point number inconsistencies
+        bounds = tuple(np.array(extent.rio.bounds()).round(8))
+        res = tuple(np.abs(extent.rio.resolution()).round(8))
+        epsg = int(str(extent.rio.crs)[5:])
+
+        # retrieve resampler
+        resampler_name = self.config["resamplers"][metadata["type"]]
+        resampler_func = getattr(rasterio.enums.Resampling, resampler_name)
+
+        # load data, i.e. fetch data from links in STAC results
+        data = stackstac.stack(
+            self.src,
+            assets=[metadata["name"]],
+            resampling=resampler_func,
+            bounds=bounds,
+            epsg=epsg,
+            resolution=res,
+            fill_value=self.config["na_value"],
+            dtype=self.config["dtype"],
+        )
+        data = data.compute(**(self.config["dask_params"] or {}))
+
+        # subset temporally
+        bounds = extent.sq.tz_convert(self.tz).time.values
+        keep = ((data.time >= bounds[0]) & (data.time < bounds[1])).values
+        data = data.sel(time=keep)
+
+        # mosaicking in case of temporal grouping
+        # convert datetimes to daily granularity - resample by day
+        def _mosaic_ints(x, axis=0, na_value=np.nan):
+            max_idx = np.argmax(x != na_value, axis=axis)
+            # handle cases where all values are na_value
+            all_na = np.all(x == na_value, axis=axis)
+            chosen = np.choose(max_idx, x)
+            # where all values are na_value, fill with na_value, else use chosen value
+            return np.where(all_na, np.full(chosen.shape, na_value, dtype=x.dtype), chosen)
+
+        if self.config["group_by_solar_day"]:
+            if len(data.time):
+                days = data.time.astype("datetime64[D]")
+                if data.dtype.kind == "f":
+                    data = data.where(data != self.config["na_value"])
+                    data = data.groupby(days).first(skipna=True)
+                else:
+                    data = data.groupby(days).reduce(_mosaic_ints, na_value=self.config["na_value"])
+                data["time"] = data.time.astype("datetime64[D]").values
```

Comment:
Handle nanosecond precision warning from pandas, there is maybe a better way for this
```suggestion
                days = data.time.astype("datetime64[ns]").dt.floor("D")
                if data.dtype.kind == "f":
                    data = data.where(data != self.config["na_value"])
                    data = data.groupby(days).first(skipna=True).rename({"floor": "time"})
                else:
                    data = data.groupby(days).reduce(_mosaic_ints, na_value=self.config["na_value"]).rename({"floor": "time"})
                data["time"] = data.time.values
```